src/Adapters/Storage/Mongo/MongoCollection.js

```diff
@@ -112,7 +112,20 @@ export default class MongoCollection {
       findOperation = findOperation.project(keys);
     }
 
-    if (caseInsensitive) {
+    if (query.username?.$regex || query.email?.$regex) {
+      const indexName = query.username?.$regex ? 'username_1' : 'email_1';
+      findOperation = await new Promise((resolve, reject) => {
+        this._mongoCollection.indexExists(indexName, (error, result) => {
```

Comment:
Yeah that `indexExists` function was hard to find. I think you're right about being able to not use a Promise wrapper. I went with creating a new Promise because that's what I saw them doing other places (specifically in the `_ensureSparseUniqueIndexInBackground`) method below. However, it does seem like it would be cleaner not to create our own Promise here.